indra/belief/__init__.py

```diff
@@ -205,6 +209,86 @@ def check_prior_probs(self, statements):
                     raise Exception(msg)
 
 
+class BayesianScorer(SimpleScorer):
+    """This is a belief scorer which assumes a Beta prior and a set of prior
+    counts of correct and incorrect instances for a given source. It exposes
+    and interface to take additional counts and update its probability
+    parameters which can then be used to calculate beliefs on a set of
+    Statements.
+
+    Parameters
+    ----------
+    prior_counts : dict
+        A dictionary of counts of the form [pos, neg] for
+        each source.
+    subtype_counts : dict
+        A dictionary of counts of the form [pos, neg] for
+        each subtype within a source.
+    """
+    def __init__(self, prior_counts, subtype_counts):
```

Comment:
I feel like this `__init__` should be able to take advantage of the `SimpleScorer`'s `__init__` via `super(BayesianScorer, self).__init__()`. It might require a bit of changing of the interface, but 3/5 of the lines here are the same as `SimpleScorer`

Reply:
The meaningful shared part of the code is in the update_probs method which does call the same method of the parent class.